### DISCUSSION TITLE
Positive endgame ideas

### DISCUSSION THREAD

**Discussion by msx80** [msx80/DoorsOfDoomOmicron#9] [Ideas]
So, i heard different players (like @APT37) that were unsatisfied with the lack of a "successful endgame" in Doors of Doom. Currently, you can go on forever and your only option at the end is to die. This is how "infinite" games usually work, but i understand the complaints.

So i'd like to discuss implementing an endgame that still retains the "infinite run" possibilities.

The first thing that comes to mind is:

- some final boss drops the Dungeon Map item (usable), representing a map where the exit is located.
- if you use the Map, you'll exit the game to the successful endgame screen
- if you don't use it, you can still keep opening doors and go on
- if you die with the Map in inventory, you still get the "die" endscreen
- after some amount of levels past the final boss, the game starts gradually pumping up the stats of monsters. This is to prevent the infinite loop trick that @krushia (and others) found in #3 enabling infinite progress.

For this to work, i would need to redesign the Wisdom object since it was ment as a kind of joke, a game breaking super powerful item to have fun after the player beat the "unbeatable" boss. I like the object and would like to retain it, perhaps with a debuff.

How does it sound ?

**Comment by APT37** [comment]
How about making Ancient fights inescapable in addition to not letting the Ancient Wisdom work against them? Additionally fixing the 1001 crash bug to ensure that even the luckiest/most skilled player will eventually die (or use the map to 'escape').

**Reply by msx80** [reply]
Inescapability surely needs to be considered as it is too easy now, you can just decide who you want to fight.

The 1000 bug can be fixed easily but there's no way the game will last that long once it's all fixed. It's meant to be a quick game, not a week long endeavour. Like 150/200 levels top.

I did some games yesterday, i reached 100 and then it was so long until 200 when the Ancient arrives, it was very boring with all the same monsters over and over. I think i'll aim at about 150 levels, 200 for very lucky runs.

Also the game is really too easy now, almost all runs i did i went past 100, with or without pants. I was able to hoard over 60 rocks which easily defeats the Ancient. The last set of monsters (GOLEM, KEYMASTER, TARDIGRADE, MAGE, VAMPIRE, DEMON, DRAGON, DEVIL) works way too good together, giving rocks, elixir/blood, scrolls, fangs and all the good stuff to cheat the system and go on forever.

**Reply by APT37** [reply]
The range from 100 to 200 is rather long indeed. I use it for farming Rocks and getting armor I don't have yet. No pants though.

**Comment by krushia** [comment]
Increasing the point cost of retreat (like each coward thingy subtracts 1.2 times the points the prior one did) is another thing that crossed my mind. With my run I could retreat an amusing number of times just to collect rocks, then get way more points by killing one demon (or is it devil? i forget lol).

**Reply by APT37** [reply]
Yeah, the slingshot is pretty broken combined with being able to pick enemies, huh.

What exactly do you mean by increasing the cost of retreat? A higher score penalty? I for one don't care about the high score, so that's pretty ineffective, but in general I agree that the current penalty is too low with how easy it is to farm points late game...

**Reply by msx80** [reply]
score penalty is ok but there needs to be something more. I was thinking maybe make escaping cost some gold. Like 10 or 20? should be enought that it's still very doable but abusing it will be limited.

**Comment by msx80** [comment]
So i'm currently considering this to balance the game:

  * create a new monster "DUNGEON BOSS", that start appearing at 150, and drops the map.
  * remove the GOLEM at around level 100/120
  * maybe add a couple of new monsters to spice up the range 100-150 (maybe a werewolf or something?)
  * the ancient will appear past 150 as an extra for people who keeps playing without exiting the dungeon (so the wisdom can remain, eventually with a bit of retouching)
  * After 150 or so, start automatically increasing the health and damage of monsters by a percentage proportional to the level number
  * award some substantial points for using the map and exiting, so people can try to adjust how much they want to push their luck past the end
  * make escaping costs money
  * debuff the crown maybe. It was meant to be like the "final item" in the game, kind of the sacred graal, but now you can basically farm them :)
  * generally make the midgame a bit harder to avoid reaching the endgame every single run.

**Reply by APT37** [reply]
Sounds good, but I disagree on one point - paying gold to escape. We're going from no actual gameplay cost (score penalty) to potentially ending a run early (no enough gold for more keys), wouldn't a chance based escape system make more sense (different chances per monster type, lower chances on higher levels)? It might also end some runs, but because the player was too weak, not because they were too poor. Or maybe it's all about dungeon capitalism? :D

**Comment by msx80** [comment]
i've actually implemented all the stuff (except escape cost) and it seems pretty balanced, still kinda easy but ok. I also some more content, and now the game save the state so you can pause at any time. Anyone interested, feel free to playtest the latest build which is available here: https://github.com/msx80/DoorsOfDoomOmicron/actions/runs/10581214269

I'm searching for a victory jingle and other minor stuff then i think i'll make a release

**Reply by home-brain** [reply]
I really enjoy this version of the game. Would it be possible to sell items at, let's say 10% of cost, to be able to get rid of the overstock of items to be used to buy keys/potions?

**Reply by APT37** [reply]
The exact percentage has to finessed, but 10-25% seem balanced to me.
